spec/lib/panoptes_client_spec.rb

```diff
@@ -0,0 +1,33 @@
+require './spec/fixtures/project_roles.rb'
+require './spec/fixtures/project.rb'
+
+RSpec.describe PanoptesClient, type: :lib do
+  include_context 'role parsing'
+  include_context 'project parsing'
+
+  let(:panoptes_client) { described_class.new(123) }
+  let(:client_double) { double }
+
+  before do
+    allow(panoptes_client).to receive(:client).and_return(client_double)
+  end
+
+  describe '#roles' do
+    let(:parsed_roles) { { 1 => ["collaborator"], 2 => ["owner"], 3 => ["researcher"] } }
+
+    it 'should return a hash' do
+      allow(client_double).to receive(:paginate).and_return(raw_roles.with_indifferent_access)
```

Comment:
If the client only returns a hash with string keys (not indifferent) could this be introducing bugs when looking up response objects with symbol keys?

If it doesn't introduce bugs, could the `raw_roles` fixture ensure `with_indifferent_access` instead of everywhere you use this? Might save some extra typing and spec bugs where it's left off and we use a symbol vs string key

Reply:
Can you control how Rspec parses the shared context? The fixtures have string keys, but after they're loaded by rspec they end up as symbols, which is why I had to add the indifferent_access bit.

Reply:
This is how ruby handles hashes now, you could add `with_indifferent_access` on the let block, e.g. here https://github.com/zooniverse/tove/blob/0809cc8488c6fef9544c9dc0be02f81abbd1a201/spec/fixtures/project_roles.rb#L22

or maybe use json to encode them across (multi line stings) but this is ugly, i'm not sure.